frontend/src/model/beta/images/UploadNewImageDialog.tsx

```diff
@@ -0,0 +1,127 @@
+import ContentCopy from '@mui/icons-material/ContentCopy'
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  IconButton,
+  Stack,
+  Tooltip,
+  Typography,
+} from '@mui/material'
+import { useTheme } from '@mui/material/styles'
+import { useGetUiConfig } from 'actions/uiConfig'
+import { useGetCurrentUser } from 'actions/user'
+import { useState } from 'react'
+import Loading from 'src/common/Loading'
+import MessageAlert from 'src/MessageAlert'
+import CodeLine from 'src/model/beta/images/CodeLine'
+
+interface UploadModelImageDialogProps {
+  open: boolean
+  handleClose: () => void
+}
+
+export default function UploadModelImageDialog({ open, handleClose }: UploadModelImageDialogProps) {
+  const { uiConfig, isUiConfigLoading, isUiConfigError } = useGetUiConfig()
+  const { currentUser, isCurrentUserLoading, isCurrentUserError } = useGetCurrentUser()
+
+  const theme = useTheme()
+
+  const [displayToken, setDisplayToken] = useState(false)
+  const [displayedToken, setDisplayedToken] = useState('')
+
+  const regenerateToken = async () => {
+    const { token } = await fetch('/api/v1/user/token', {
+      method: 'POST',
+    }).then((res) => res.json())
+
+    return token
+  }
+
+  const showToken = async () => {
+    const token = await regenerateToken()
```

Comment:
Given that `regenerateToken()` can error, we should also ensure that `token` is valid before setting it.
```
if (token) {
    setDisplayedToken(token)
    setDisplayToken(true)
}
```